src/lib/adapters/firebase/index.ts

```diff
@@ -301,11 +305,28 @@ export default class FirebaseAdapter implements Adapter {
 			decimals: 18,
 			amount: 7843900000000000000000n.toString(),
 			image: 'https://s2.coinmarketcap.com/static/img/coins/64x64/4943.png',
+			address: '0x0000000000000000000000000000000000000000',
 		}
 
 		setDoc(daiDoc, daiData, { merge: true })
 	}
 
+	async updateBalance(address: string, token: Token): Promise<void> {
+		if (token.address) {
+			// only native token is supported for now
+			return
+		}
+
+		const nativeTokenAmount = await getBalance(address)
+
+		const ethDoc = doc(db, `users/${address}/balances/${token.symbol.toLowerCase()}`)
+		const ethData: { amount: string } = {
+			amount: nativeTokenAmount.toString(),
+		}
+
+		setDoc(ethDoc, ethData, { merge: true })
```

Comment:
Why are we storing this?

Reply:
I opened an issue about this topic: https://github.com/logos-innovation-lab/waku-objects-playground/issues/171